src/wallet/scriptpubkeyman.cpp

```diff
@@ -1460,7 +1460,6 @@ bool LegacyScriptPubKeyMan::TopUpInner(unsigned int kpSize)
                     m_storage.UpdateProgress(strMsg, static_cast<int>(dProgress));
                 }
             }
-            if (ShutdownRequested()) break;
```

Comment:
this makes it non-interruptible... could maybe add another helper in `WalletStorage`/`CWallet` that would return `chain().shutdownRequested()` and use it here

Reply:
done through WalletStorage interface.